chef/cookbooks/glance/recipes/ceph.rb

```diff
@@ -17,42 +17,57 @@
 # limitations under the License.
 #
 
+ceph_conf = node[:glance][:rbd][:store_ceph_conf]
+admin_keyring = node[:glance][:rbd][:store_admin_keyring]
+
 ceph_env_filter = " AND ceph_config_environment:ceph-config-default"
 ceph_servers = search(:node, "roles:ceph-osd#{ceph_env_filter}") || []
 if ceph_servers.length > 0
   include_recipe "ceph::keyring"
 else
+
   # If external Ceph cluster will be used,
   # we need install ceph client packages
-  if node[:platform] == "suse" && File.exists?("/etc/ceph/ceph.client.admin.keyring")
+  if node[:platform] == "suse"
     package "ceph-common"
-    package "python-ceph"
-  else 
+  end
+
+  if File.exists?(admin_keyring)
+    Chef::Log.info("Using external ceph cluster for glance, with automatic setup.")
+  else
+    Chef::Log.info("Using external ceph cluster for glance, with no automatic setup.")
     return
   end
+
```

Comment:
(I don't feel the empty line is needed, so I'd drop it unless you strongly care about it)